src/core/projection.ts

```diff
@@ -121,7 +121,17 @@ export default class Projection {
             this.radii.minor = radius || this.radii.major || this.baseRadius
     }
 
-    invertY = (y: number, z: number): number => Math.pow(2, z) - 1 - y
+    invertY = (y: number, z: number): number => {
+        const b = this.crs.projection.bounds
+        if (b === null || b.min.y === b.max.y || !isFinite(b.min.y)) {
+            // Map uses default projection and/or has unspecified/infinite bounds
+            return Math.pow(2, z) - y

```

Comment:
Much better! Add back in the `- 1` here and then I'll merge it in.

Reply:
It might also be good to use `if(this.tileMapResource.crsCode === 'EPSG:4326')` instead since 4326 is mercator/the default projection.

Reply:
Definitely agree that the condition as currently written is a bit silly - I was trying to capture any case where the bounds might not be defined, including other projections with similar bounds (if any exist) or infinite generated worlds. If you expect EPSG:4326 to be the only such case, I'll definitely make that change.
(Don't know when the `- 1` disappeared, will put that back)